GoogleUtilities/AppDelegateSwizzler/Private/GULAppDelegateSwizzler.h

```diff
@@ -43,49 +43,54 @@ typedef NSString *const GULAppDelegateInterceptorID;
+ *  After calling this method the following App Delegate methods will be proxied in addition to
+ *  the methods proxied by proxyOriginalDelegateWithoutRemoteNotificationMethods:
  *  @code
  *    - application:didRegisterForRemoteNotificationsWithDeviceToken:
  *    - application:didFailToRegisterForRemoteNotificationsWithError:
  *    - application:didReceiveRemoteNotification:fetchCompletionHandler:
  *    - application:didReceiveRemoteNotification:
  *  @endcode
  *
- *  To proxy these methods use +[GULAppDelegateSwizzler
- *  proxyOriginalDelegateRemoteNotificationMethods]. The methods have to be proxied separately to
- *  avoid potential warnings from Apple review about missing Push Notification Entitlement (e.g.
- *  https://github.com/firebase/firebase-ios-sdk/issues/2807)
- *
  *  The method has no effect for extensions.
  *
- *  @see proxyOriginalDelegateRemoteNotificationMethods
+ *  NOTE: If your application does not use APNS, you should consider using
+ *  + [GULAppDelegateSwizzler proxyOriginalDelegateWithoutRemoteNotificationMethods] instead to
+ *  avoid risk of warnings from Apple review about missing Push Notification Entitlement (like
+ *  https://github.com/firebase/firebase-ios-sdk/issues/2807)
+ *
+ *  @see proxyOriginalDelegateWithoutRemoteNotificationMethods
  */
 + (void)proxyOriginalDelegate;
 
-/** This method ensures that the original app delegate has been proxied including APNS related
- *  methods. Call this before registering your interceptor. This method is safe to call multiple
- *  times (but it only proxies the app delegate once) or
- *  after +[GULAppDelegateSwizzler proxyOriginalDelegate]
+/** This method ensures that the original app delegate has been proxied. Call this before
+ *  registering your interceptor. This method is safe to call multiple times (but it only proxies
+ *  the app delegate once).
  *
- *  This method calls +[GULAppDelegateSwizzler proxyOriginalDelegate] under the hood.
- *  After calling this method the following App Delegate methods will be proxied in addition to
- *  the methods proxied by proxyOriginalDelegate:
+ *  This method doesn't proxy APNS related methods:
  *  @code
  *    - application:didRegisterForRemoteNotificationsWithDeviceToken:
  *    - application:didFailToRegisterForRemoteNotificationsWithError:
  *    - application:didReceiveRemoteNotification:fetchCompletionHandler:
  *    - application:didReceiveRemoteNotification:
  *  @endcode
  *
+ *  To proxy these methods use +[GULAppDelegateSwizzler
+ *  proxyOriginalDelegate]. The methods have to be proxied separately to
+ *  avoid potential warnings from Apple review about missing Push Notification Entitlement (e.g.
+ *  https://github.com/firebase/firebase-ios-sdk/issues/2807)
+ *
  *  The method has no effect for extensions.
  *
  *  @see proxyOriginalDelegate
  */
-+ (void)proxyOriginalDelegateRemoteNotificationMethods;
```

Comment:
You will have to make a change in Analytics as a result of this. Is this going to be a major version bump>

Reply:
@paulb777 Do you think we can update the analytics as a part of Copybara CL?

Reply:
Because this is blocking the release, I would vote to revert the renaming, and revisit it after the release is out

Reply:
This breaks fcm test app for extension, its better to fix this if its going out for breaking change release anyway.